lib/IRGen/GenType.cpp

```diff
@@ -1797,6 +1797,57 @@ static bool isIRTypeDependent(IRGenModule &IGM, NominalTypeDecl *decl) {
   }
 }
 
+bool irgen::mightContainMetadata(const IRGenModule &IGM, const CanType type) {
```

Comment:
What does it mean to "contain metadata"? It looks like there should be existing abstractions here that can answer this question.

Reply:
What I really mean is does it contain metadata that I care about emitting type metadata ref for in the caller / pass on to the callee. which is not always the case. I will add a comment to explain that

Reply:
Can you explain what the condition does without looking at specific cases?

Reply:
I am not sure I understand your question, the function makes sure we actually need to create metadata and if the type actually contains dependent types.

Reply:
Ok, but this sort of operation happens all over IRGen, why do you need a new function for it?

Reply:
There are very similar operations in IRGen, yes, I used the IR Dependent type operation in it for example. not exactly what I wanted here

Reply:
We need to sit down and figure out what the right check is.